src/sources/socket/tcp.rs

```diff
@@ -206,28 +209,29 @@ impl TcpSource for RawTcpSource {
                     now,
                 );
 
-                let host_key_path = self.config.host_key.as_ref().map_or_else(
-                    || [BorrowedSegment::from(log_schema().host_key())],
-                    |key| [BorrowedSegment::from(key)],
+                let legacy_host_key = self.config.host_key.as_ref().map_or_else(
+                    || parse_value_path(log_schema().host_key()).ok(),
+                    |k| k.path.clone(),
                 );
 
                 self.log_namespace.insert_source_metadata(
                     SocketConfig::NAME,
                     log,
-                    Some(LegacyKey::InsertIfEmpty(&host_key_path)),
+                    legacy_host_key.as_ref().map(LegacyKey::InsertIfEmpty),
                     path!("host"),
                     host.ip().to_string(),
                 );
 
-                let port_key_path = self.config.port_key.as_ref().map_or_else(
-                    || [BorrowedSegment::from("port")],
-                    |key| [BorrowedSegment::from(key)],
-                );
+                let legacy_port_key = self
+                    .config
+                    .port_key
+                    .as_ref()
+                    .map_or_else(|| parse_value_path("path").ok(), |k| k.path.clone());
```

Comment:
Shouldn't this be

```suggestion
                    .map_or_else(|| parse_value_path("port").ok(), |k| k.path.clone());
```

? :thinking:

Reply:
I don't even know what to tell you 🤦